base/threadingconstructs.jl

```diff
@@ -22,14 +22,14 @@ See also: `BLAS.get_num_threads` and `BLAS.set_num_threads` in the
 """
 nthreads() = Int(unsafe_load(cglobal(:jl_n_threads, Cint)))
 
-function threading_run(func)
-    ccall(:jl_enter_threaded_region, Cvoid, ())
+function threading_run(func, static)
+    static && ccall(:jl_enter_threaded_region, Cvoid, ())
```

Comment:
Is `jl_enter_threaded_region` really just for static scheduling? It seems like it affects I/O (allowing the libuv loop to be run on the non-primary thread), as well as the sleep/wake logic in PARTR. Maybe this deserves its own flag, like `prioritize_io`?

Reply:
Yeah, the threaded region is an orthogonal concept to what `@threads :static` needs to offer (though I think it's fine to keep the behavior of `:static`). But IMHO it's better to expose this as an orthogonal API like `prioritize_io() do; $user_code; end` https://github.com/JuliaLang/julia/pull/43420#discussion_r769999468

Reply:
Yes, I think for now we can just call jl_enter_threaded_region unconditionally here.

Reply:
Ok. updated